example/glue/VerboseManagerImpl.cpp

```diff
@@ -61,22 +61,21 @@ bool
 MM_VerboseManagerImpl::configureVerboseGC(OMR_VM *omrVM, char *filename, uintptr_t fileCount, uintptr_t iterations)
 {
 	OMRPORT_ACCESS_FROM_OMRVM(omrVM);
-	if (MM_VerboseManager::configureVerboseGC(omrVM, filename, fileCount, iterations)) {
-		this->fileCount = fileCount;
-		this->iterations = iterations;
-		size_t len = strlen(filename);
-		this->filename = (char *)omrmem_allocate_memory(len+1, OMRMEM_CATEGORY_MM);
-		strncpy(this->filename, filename, len);
-		this->filename[len] = '\0';
-		return true;
-		if (NULL != this->filename) {
-			return false;
-			strncpy(this->filename, filename, len);
-			this->filename[len] = '\0';
-			return true;
-		}
+	if (!MM_VerboseManager::configureVerboseGC(omrVM, filename, fileCount, iterations)) {
+		return false;
+	}
+	this->fileCount = fileCount;
+	this->iterations = iterations;
+	size_t len = strlen(filename);
+
+	this->filename = (char *)omrmem_allocate_memory(len+1, OMRMEM_CATEGORY_MM);
+	if (NULL == this->filename) {
+		return false;
 	}
-	return false;
+
+	strcpy(this->filename, filename);
```

Comment:
can we use strncpy since we do have the length?  It will stop some of the static analysis tooling from complaining. If you think it really does not make sense we can leave it as is.

Reply:
GCC is complaining about this in my version.  The complaint in question is that we are using the length of the source buffer as an argument meant to be the length of the destination buffer.    Although `len` is technically the correct length of the destination, GCC can tell we are using the result of `strlen(filename)`.  The current advice for this issue is to just use `strcpy`, since it is guaranteed to work in this situation.

It seems with these new warnings that GCC is trying to look a little too hard into how these functions were meant to be used.  Our usage is completely valid, but `strncpy` was originally meant to copy non-null-terminated strings.  I have run into other warnings where we use `strncpy` to make sure that we don't copy the `\0` off the end of a source string, and it produces a warning that complains we are truncating the source string.  Instead, I have switched instances of that to use `strcat`.